couchpotato/core/media/_base/providers/torrent/hd4free.py

```diff
@@ -25,10 +25,11 @@ class Base(TorrentProvider):
     def _search(self, movie, quality, results):
         data = self.getJsonData(self.urls['search'] % (self.conf('apikey'), self.conf('username'), getIdentifier(movie), self.conf('internal_only')))
 
-        if data:
-            if self.login_fail_msg in data['error']: # Check for login failure
-                self.disableAccount()
-                return
+        if 'error' in data:
+            if data:
```

Comment:
This commit needs a fix. it most likely introduces the same type of bug its meant to fix. 

lines 28 and 29 need to be reversed because you shouldn't be checking for "error" in "data" before you know that
"data" dict exists first. It should be as below.

    if data:
        if 'error' in data: